src/ipc/ipc4/handler.c

```diff
@@ -52,6 +52,10 @@ struct ipc4_msg_data msg_data;
 /* fw sends a fw ipc message to send the status of the last host ipc message */
 struct ipc_msg msg_reply;
 
+static bool ipc_cmd_in_progress;
+
+static struct ipc_msg msg_notify;
```

Comment:
In theory we could use "msg_reply", but I was 100% sure this is safe, so added a separate object for the notifies. @ujfalusi @RanderWang @mmaka1 ?

Reply:
In theory you could send the notification in between a message reception and sending back the reply, but why do you need this a global?

Reply:
msg_reply is for host msg, notification is for dsp msg, so we need separate msg. we don't need global msg for notification. There are a few types of notification and we can create msg for each type.